app/models.py

```diff
@@ -41,6 +41,32 @@ def to_json(self):
 
         return json_song
 
+    def get_related_songs_json(self, top=None):
+        # default to pulling top 10 results
+        if top is None:
+            top = 10
+        query_str = '''
+            select rel.id, rel.title, rel.url, rel.artist, rel.created, rel.user_id, count(srv) as vote_count
+            from songs as s
+            join song_relations as sr
+                on s.id = sr.song1_id
+                or s.id = sr.song2_id
+            join song_relation_votes as srv
+                on srv.song_relation_id = sr.id
+            join songs as rel
+                on (rel.id = sr.song1_id and rel.id <> sr.song2_id)
+                or (rel.id = sr.song2_id and rel.id <> sr.song1_id)
+            where s.id = {0} and rel.id <> {0}
+            group by rel.id, rel.title, rel.url, rel.artist, rel.created, rel.user_id
+            order by vote_count desc
+            limit {1}
+        '''.format(self.id, top)
```

Comment:
Rofl dude this is a sql injection vulnerability. Top comes from query args and you don't ever check if it's an int.